frame/staking/src/lib.rs

```diff
@@ -1646,19 +1638,31 @@ impl<T: Trait> Module<T> {
 	}
 }
 
-impl<T: Trait> pallet_session::OnSessionEnding<T::AccountId> for Module<T> {
-	fn on_session_ending(_ending: SessionIndex, start_session: SessionIndex) -> Option<Vec<T::AccountId>> {
+impl<T: Trait> pallet_session::SessionManager<T::AccountId> for Module<T> {
+	fn new_session(new_index: SessionIndex) -> Option<Vec<T::AccountId>> {
 		Self::ensure_storage_upgraded();
-		Self::new_session(start_session - 1).map(|(new, _old)| new)
+		if new_index == 0 {
```

Comment:
we need to provide the validator set for the first session, this doesn't change the behavior of staking module.

Reply:
But I think we have some issue there, the first era is not completely set as we should also write:
```rust
		CurrentEraStartSessionIndex::put(session_index);
		CurrentEraStart::put(T::Time::now());
		BondedEras::mutate(|bonded| bonded.push((current_era, 0)));
```
But on then I'm stuck because T::Time::now return 0 if called before the first inherent that set timestamp, thus the first era does start from moment 0 which makes its duration very long. Maybe we can not do reward first the era that started at moment 0.

Would it be ok not to reward such eras ?